3.6 update docs for XRootD (#964)
opensciencegrid/docs: docs/data/xrootd/xrootd-authorization.md

will map FQANs starting with `/osg/` to the user `osg01`.

See the [VOMS Mapping documentation](https://github.com/xrootd/xrootd/tree/master/src/XrdVoms#voms-mapping) for details.

> VOMS Mapfiles previously used with LCMAPS should continue to work unmodified,
> but the plugin can only look at a single mapfile,
> so if you want to use the mappings provided at `/usr/share/osg/voms-mapfile-default`
> (by the `vo-client-lcmaps-voms` package),
> you will have to copy them to `/etc/grid-security/voms-mapfile`.

Reviewer: ```suggestion
VOMS Mapfiles previously used with LCMAPS should continue to work unmodified,
but the plugin can only look at a single mapfile,
so if you are using the mappings provided in `/usr/share/osg/voms-mapfile-default`
(by the `vo-client-lcmaps-voms` package),
you will have to copy them to `/etc/grid-security/voms-mapfile`.
```

        [938.1kB/938.1kB][100%][==================================================][938.1kB/s]

    If your transfer does not succeed, re-run `xrdcp`  with `--debug 2` for more information.

Updating to OSG 3.6
--------------------

There are some manual steps that need to be taken for authentication to work in OSG 3.6.

### Ensure OSG XRootD packages are fully up-to-date

Some authentication configuration is provided by OSG packaging.
Old versions of the packages may result in broken configuration.
It is best if your packages match the versions in the appropriate `release` subdirectories of
<https://repo.opensciencegrid.org/osg/3.6/>, but at the very least these should be true:

-   `xrootd >= 5.4`
-   `xrootd-multiuser >= 2` (if using multiuser)
-   `xrootd-scitokens >= 5.4` (if using SciTokens/WLCG Tokens)
-   `xrootd-voms >= 5.4.2-1.1` (if using VOMS auth)
-   `osg-xrootd >= 3.6`
-   `osg-xrootd-standalone >= 3.6` (if installed)

> -   `xcache >= 3` (if using xcache-derieved software such as stash-cache, stash-origin, atlas-xcache, or cms-xcache)

Reviewer: ```suggestion
-   `xcache >= 3` (if using xcache-derived software such as stash-cache, stash-origin, atlas-xcache, or cms-xcache)
```

    otherwise XRootD may fail to start.**

-   If you do not have both an unauthenticated stash-cache and an authenticated stash-cache on the same server,
    uncomment `set EnableVoms = 1` in `/etc/xrootd/config.d/10-osg-xrdvoms.cfg`.

-   If you have both an an authenticated stash-cache and an unauthenticated stash-cache on the same server,
    add the following block to `/etc/xrootd/config.d/10-osg-xrdvoms.cfg`:

        if named stash-cache-auth
            set EnableVoms = 1
        fi

-   If you are using XRootD Multiuser, create a VOMS Mapfile at `/etc/grid-security/voms-mapfile`,
    with the syntax [described above](#mapping-voms-attributes-to-users),
    then add `voms.mapfile /etc/grid-security/voms-mapfile` to your XRootD config if it's not already present.

!!! note
    In order to make `yum update` easier, `xrootd-lcmaps` has been replaced with an empty package, which can be removed after upgrading.

#### Updating your authorization database

Unlike the XrdLcmaps plugin, which mapped VOMS FQANs to users `u`, the XrdVoms plugin maps FQANs to
groups `g`, roles `r`, and organizations `o`, as described in the [mapping VOMS attributes section](#mapping-voms-attributes).

> You can still [use a VOMS mapfile](#mapping-voms-attributes-to-users)
> but if you want to use the mappings provided at `/usr/share/osg/voms-mapfile-default`
> by the `vo-client-lcmaps-voms` package,
> you will have to copy them to `/etc/grid-security/voms-mapfile`.

Reviewer: ```suggestion
You can still [use a VOMS mapfile](#mapping-voms-attributes-to-users)
but if you want to use the mappings provided at `/usr/share/osg/voms-mapfile-default`
by the `vo-client-lcmaps-voms` package,
you must copy them to `/etc/grid-security/voms-mapfile`.
```

Replace mappings based on users with mappings based on the other attributes.
For example, instead of
```
u uscmslocal /uscms rl
```
use
```
g /cms/uscms /uscms rl
```

> If you need to make a mapping based on group _and_ role, create an use a "compound ID" as described in

Reviewer: ```suggestion
If you need to make a mapping based on group _and_ role, create and use a "compound ID" as described in
```
